### PR TITLE
Adding Twotrees Full range printer profiles

### PR DESCRIPTION
**Pull request by Lester3DGadgets**
Profile for:
Bluer / Bluer Plus / Sapphire Pro / Sapphire Plus

Nozzle size : 0.2mm to 1mm

Material: PLA / ABS / PETG / TPU

### CODE REVIEW

**Comment by Lester3DGadgets** [comment]
Wondering why keep failing on check

**Comment by konskarm** [comment]
Thanks for your PR @Lester3DGadgets. 

One thing I notice, the Two Trees Bluer machine definition already exist in our codebase. It was added in PR https://github.com/Ultimaker/Cura/pull/8620. In addition, there is another PR before yours (https://github.com/Ultimaker/Cura/pull/9102) that is attempting to add the Two Trees Sapphire Pro definition. I would like to request from you to contact @WashingtonJunior and @G4Cab and make sure they are ok with you overridding their machine profiles. If they are fine with it, just make sure you also remove those old profiles as part of your PR.

**Note:** If it is decided that your TwoTrees Bluer will override the old one, we need to add it to our version upgrade script, since the name of the printer definition file is not the same (`twotrees_bluer.def.json` (old) vs `two_trees_bluer.def.json`(new)). This will be done on our side, but I am mentioning it here so that we don't forget.

**Comment by Lester3DGadgets** [comment]
Hello @WashingtonJunior @G4Cab,

Can we standardize and use the profile that i created? It structured base profile and included all twotrees models such as Bluer, Bluer Plus, Sapphire Pro and Saspphire. On top of that the profile also included few qualities selection with a few material like PLA, ABS, PETG and TPU.

**Comment by konskarm** [comment]
Let's wait for a week or so and if there is no response, I'll go forward with yours.

**Comment by G4Cab** [comment]
Sure, do what you need to do, i only created this profile, cause it didn't existed

**Comment by konskarm** [comment]
Thanks for your response @G4Cab. I just needed confirmation. I'll go forward and close your other two PRs https://github.com/Ultimaker/Cura/pull/9102 and https://github.com/Ultimaker/Cura/pull/9132, and will continue with this one.

File: resources/definitions/two_trees_base.def.json
Comment (konskarm) [inline resources/definitions/two_trees_base.def.json:81]:
```suggestion
        "retraction_speed": { "default_value" : 40 },
```

File: resources/definitions/two_trees_base.def.json
Comment (konskarm) [inline resources/definitions/two_trees_base.def.json:88]:
```suggestion
        "cool_fan_full_layer": { "value": "2" },
```

File: resources/definitions/two_trees_bluer.def.json
Comment (konskarm) [inline resources/definitions/two_trees_bluer.def.json:6]:
```suggestion
        "machine_start_gcode": { "default_value": "G28 ;Home\nG92 E0 ;Reset Extruder\nG1 Z2.0 F3000 ;Move bed down\nG1 X10.1 Y20 Z0.28 F5000.0 ;Move to start position\nG1 X10.1 Y200.0 Z0.28 F1500.0 E15 ;Draw the first line\nG1 X10.4 Y200.0 Z0.28 F5000.0 ;Move to side a little\nG1 X10.4 Y20 Z0.28 F1500.0 E30 ;Draw the second line\nG92 E0 ;Reset Extruder\nG1 Z2.0 F3000 ;Move Bed up" },
```

File: resources/definitions/two_trees_bluer.def.json
Comment (konskarm) [inline resources/definitions/two_trees_bluer.def.json:17]:
```suggestion
        [-47, 37],
        [-47, -23],
        [47, -23],
        [47, 37]
        ]
        },
```

File: resources/definitions/two_trees_bluer.def.json
Comment (konskarm) [inline resources/definitions/two_trees_bluer.def.json:23]:
```suggestion
        "speed_print": { "value": 50.0 },
        "speed_travel": { "value": 180 },
        "speed_travel_layer_0": { "value": 100 },
        "retraction_speed": { "default_value" : 50 },
```

File: resources/definitions/two_trees_bluer.def.json
Comment (konskarm) [inline resources/definitions/two_trees_bluer.def.json:29]:
```suggestion
        "material_flow": { "value": 90 },
        "infill_material_flow": { "value": 100 },
        
        "infill_overlap": { "value": 30 },
```

File: resources/definitions/two_trees_bluer.def.json
Comment (konskarm) [inline resources/definitions/two_trees_bluer.def.json:35]:
```suggestion
        "platform": "twotrees235x235_generic.stl"
```

File: resources/definitions/two_trees_bluerplus.def.json
Comment (konskarm) [inline resources/definitions/two_trees_bluerplus.def.json:16]:
```suggestion
        [-47, 37],
        [-47, -23],
        [47, -23],
        [47, 37]
        ]
        },
```

File: resources/definitions/two_trees_bluerplus.def.json
Comment (konskarm) [inline resources/definitions/two_trees_bluerplus.def.json:22]:
```suggestion
        "speed_print": { "value": 50.0 },
        "speed_travel": { "value": 180 },
        "speed_travel_layer_0": { "value": 100 },
        "retraction_speed": { "default_value" : 50 },
```

File: resources/definitions/two_trees_bluerplus.def.json
Comment (konskarm) [inline resources/definitions/two_trees_bluerplus.def.json:28]:
```suggestion
        "material_flow": { "value": 90 },
        "infill_material_flow": { "value": 100 },
        
        "infill_overlap": { "value": 30 },
```

File: resources/definitions/two_trees_bluerplus.def.json
Comment (konskarm) [inline resources/definitions/two_trees_bluerplus.def.json:34]:
```suggestion
        "platform": "twotrees300x300_generic.stl"
```

File: resources/definitions/two_trees_sapphireplus.def.json
Comment (konskarm) [inline resources/definitions/two_trees_sapphireplus.def.json:16]:
```suggestion
        [-30,19],
        [30,19],
        [30,-49],
        [-30,-49]
        ] 
        },
```

File: resources/definitions/two_trees_sapphireplus.def.json
Comment (konskarm) [inline resources/definitions/two_trees_sapphireplus.def.json:23]:
```suggestion
        "speed_travel": { "value": 180 },
        "speed_travel_layer_0": { "value": 100 },
        "retraction_speed": { "default_value" : 40 },
```

File: resources/definitions/two_trees_sapphireplus.def.json
Comment (konskarm) [inline resources/definitions/two_trees_sapphireplus.def.json:29]:
```suggestion
        "material_flow": { "value": 90 },
        "infill_material_flow": { "value": 100 },
        
        "infill_overlap": { "value": 30 },
```

File: resources/definitions/two_trees_sapphireplus.def.json
Comment (konskarm) [inline resources/definitions/two_trees_sapphireplus.def.json:36]:
```suggestion
        "platform": "sapphireplus_platform.stl"
        
```

File: resources/definitions/two_trees_sapphirepro.def.json
Comment (konskarm) [inline resources/definitions/two_trees_sapphirepro.def.json:6]:
```suggestion
        "machine_start_gcode": { "default_value": "G28 ;Home\nG92 E0 ;Reset Extruder\nG1 Z2.0 F3000 ;Move bed down\nG1 X10.1 Y20 Z0.28 F5000.0 ;Move to start position\nG1 X10.1 Y200.0 Z0.28 F1500.0 E15 ;Draw the first line\nG1 X10.4 Y200.0 Z0.28 F5000.0 ;Move to side a little\nG1 X10.4 Y20 Z0.28 F1500.0 E30 ;Draw the second line\nG92 E0 ;Reset Extruder\nG1 Z2.0 F3000 ;Move Bed up" },
```

File: resources/definitions/two_trees_sapphirepro.def.json
Comment (konskarm) [inline resources/definitions/two_trees_sapphirepro.def.json:18]:
```suggestion
        [-30,19],
        [30,19],
        [30,-49],
        [-30,-49]
        ] 
        },

```

File: resources/definitions/two_trees_sapphirepro.def.json
Comment (konskarm) [inline resources/definitions/two_trees_sapphirepro.def.json:24]:
```suggestion
        "speed_travel": { "value": 180 },
        "speed_travel_layer_0": { "value": 100 },
        "retraction_speed": { "default_value" : 40 },
```

File: resources/definitions/two_trees_sapphirepro.def.json
Comment (konskarm) [inline resources/definitions/two_trees_sapphirepro.def.json:30]:
```suggestion
        "material_flow": { "value": 90 },
        "infill_material_flow": { "value": 100 },
        
        "infill_overlap": { "value": 30 },
```

File: resources/definitions/two_trees_sapphirepro.def.json
Comment (konskarm) [inline resources/definitions/two_trees_sapphirepro.def.json:36]:
```suggestion
        "platform": "sapphirepro_platform.stl"
```

**Review by konskarm — CHANGES_REQUESTED** [review]
Just a few code-style changes:

- There is a mix of tabs and spaces. I highlighted all of them so that it is easy for you to find and change.
- Please rename `resources/meshes/sapphireplus_platform.stl` to `resources/meshes/two_trees_sapphireplus_platform.stl` and `resources/meshes/sapphireplus_platform.stl` to `resources/meshes/two_trees_sapphireplus_platform.stl`. It will be easier to understand to which printers they refer to.

I also noticed that the stl files seem to be empty, but once I try to download them they are properly filled, so this is probably a bug on GitHub's side.

**Comment by Lester3DGadgets** [comment]
updated all the code as per suggestion. Thanks guys.

**Comment by MarioGK** [comment]
Any idea when this might be merged?

**Comment by Lester3DGadgets** [comment]
Cura 4.9 released, unfortunately this full list of profile are not being merged  yet.

**Comment by MarioGK** [comment]
@Lester3DGadgets Thank you for the hard work, the profiles are working beautifully for me, i am using your fork for now i hope Cura maintainers could merge it soon so it would be easier, btw it seems that you have a duplicate printer in there, on the menu to add a printer there's "Two Trees" and "TwoTrees"

**Comment by Ghostkeeper** [comment]
Yes, the TwoTrees Bluer printer is duplicating the one added in November: https://github.com/Ultimaker/Cura/pull/8620

File: resources/definitions/two_trees_bluer.def.json
Comment (Ghostkeeper) [inline resources/definitions/two_trees_bluer.def.json:2]: This profile duplicates the TwoTrees Bluer printer already added to Cura. Could you update the existing definition instead? Changing the definition ID would require a version upgrade.

**Review by Ghostkeeper — CHANGES_REQUESTED** [review]
Indeed, the Two Trees Bluer printer duplicates the one added in November: https://github.com/Ultimaker/Cura/pull/8620

The previous review caught most issues already, but it wasn't noticed then that this would be duplicate.

Developers, this feature is tracked in our tracker as CURA-8212.

**Comment by Ghostkeeper** [comment]
I'm told that another developer has been working on a version upgrade to help transition users of the existing Bluer profile to your new ones. This is necessary anyway because your new one has different default profiles.

**Comment by konskarm** [comment]
Related version upgrader PR: https://github.com/Ultimaker/Cura/pull/9805.

**Pull request merged**
